packages/nextjs/hooks/scaffold-stark/useScaffoldContract.ts

```diff
@@ -10,18 +11,32 @@ export const useScaffoldContract = <TContractName extends ContractName>({
 }) => {
   const { data: deployedContractData, isLoading: deployedContractLoading } =
     useDeployedContractInfo(contractName);
+  
   const { provider: publicClient } = useProvider();
-  let contract = undefined;
-  if (deployedContractData) {
-    contract = new Contract(
-      [...deployedContractData.abi],
+
+  const contract = useMemo(() => {
+    if (!deployedContractData) return undefined;
+
+    const contractInstance = new Contract(
+      deployedContractData.abi as Abi,
       deployedContractData.address,
-      publicClient,
+      publicClient
```

Comment:
should involve `WalletClient` or `Account` instance since `publicClient` cannot sign messages.